File: src/java/product-api-container/src/main/java/com/product/api/container/core/ProductCreatedEvent.java
/*
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2024 Datadog, Inc.
 */

package com.product.api.container.core;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@JsonIgnoreProperties(ignoreUnknown = true)
public class ProductCreatedEvent {
    private final String productId;
    private final String name;
    private final Double price;

    public ProductCreatedEvent() {
        this.productId = "";
        this.name = "";
        this.price = -1.0;
    }

    public ProductCreatedEvent(String productId, String name, Double price) {
        this.productId = productId;
        this.name = name;
        this.price = price;
    }

    public String getProductId() {
        return productId;
    }

    public String getName() {
        return name;
    }

    public Double getPrice() {
        return price;
    }
}

File: src/order-service/src/Product.Api/ProductApi.Core/ProductCreatedEvent.cs
// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
// This product includes software developed at Datadog (https://www.datadoghq.com/).
// Copyright 2024 Datadog, Inc.

namespace ProductApi.Core;

public record ProductCreatedEvent
{
    public ProductCreatedEvent(Product product)
    {
        this.ProductId = product.ProductId;
        this.Name = product.Details.Name;
        this.Price = product.Details.Price;
    }
    
    public string ProductId { get; set; }

    public string Name { get; set; }

    public decimal Price { get; set; }
}